src/Integration/LocalServices/SolutionBindingInformationProvider.cs

```diff
@@ -45,6 +49,7 @@ public SolutionBindingInformationProvider(IServiceProvider serviceProvider)
         #region ISolutionBindingInformationProvider
         public bool IsSolutionBound()
         {
+            // TODO: CM2: consider changing this method to "IsLegacySolutionBound"
```

Comment:
I've added to-dos for places that are legacy-mode specific.
Later on, once _IConfigurationProvider_ returns the mode (standalone, legacy connected, connected) I'll add assertions/ifs to make sure this code only runs when in legacy mode.